### PR TITLE
#1424 - Overapproximate the interval matrix map of a zonotope

### LINKED ISSUE #1424: Overapproximate the interval matrix map of a zonotope
See Theorem 4 in http://mediatum.ub.tum.de/doc/1289985/728488.pdf.

Note: We want to use the package `IntervalMatrices`.

### PR DESCRIPTION
**Pull request by schillic**
Closes #1424.

Requires:
* ~https://github.com/JuliaReach/IntervalMatrices.jl/issues/14~
* external fix with `IntervalArithmetic`  (https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/317)

### CODE REVIEW

File: src/Approximations/overapproximate.jl [hunk 1]
```diff
@@ -809,6 +809,74 @@ end
...
+
+```math
+    v_j = \\begin{cases} 0 & i ≠ j \\\\
+          \\hat{M}_j (|c| + \\sum_{k=1}^m |g_k|) & i = j. \\end{cases}
+```
+
+[1] Althoff, Stursberg, Buss. Reachability analysis of linear systems with
+uncertain parameters and inputs. CDC 2007.
+"""
+function overapproximate(lm::LinearMap{N, <:Zonotope{N}, NM,
```
Comment (mforets) [inline src/Approximations/overapproximate.jl]: Let's use the `AbstractZonotope` interface here, so we can also pass `Hyperrectangle`. Below we have to use `genmat` etc.

**Comment by mforets** [comment]
> external fix with IntervalArithmetic

if you know a fix, let's add it here and mark it as `# patch for .... IntervalArithmetic#...` so this issue is not stuck

**Comment by schillic** [comment]
Nope, I do not even understand what is going on there.

**Comment by schillic** [comment]
> if you know a fix, let's add it here and mark it as # patch for .... IntervalArithmetic#... so this issue is not stuck

Done.

**Pull request merged**